tests/platform_tests/test_link_down.py

```diff
@@ -0,0 +1,190 @@
+            logger.info("Interface {} on fanout {} (os type {}) map to interface {} on duthost {}".format(fanout_port, fanout.hostname, fanout.get_fanout_os(), port, duthost.hostname))
+            if fanout in fanout_and_ports.keys():
+                fanout_and_ports[fanout].append(fanout_port)
+            else:
+                fanout_and_ports[fanout] = [fanout_port]
+    return fanout_and_ports
+    
+
+def is_link_down(fanout, port):
+    """
+        Either oper/admin status is down meaning link is down
+    """
+    logger.info("Checking interface {} down status on fanout host {}".format(port, fanout.hostname))
+    return fanout.is_intf_status_down(port)
+
+def is_link_up(fanout, port):
+    logger.info("Checking interface {} up status on fanout host {}".format(port, fanout.hostname))
+    return not fanout.is_intf_status_down(port)
+    
+def link_status_on_host(duthost, localhost, fanouts_and_ports, up = True):
+    for fanout, ports in fanouts_and_ports.items():
+        for fanout_port in ports:
+            # Note that there is case that fanout host not working: svcstr-7260cx3-acs-2 (not working to be RMA'ed)
+            # In that case the following assert will be thrown too.
+            if up:
+                # Make sure interfaces are up on fanout hosts
+                pytest_assert(wait_until(MAX_TIME_TO_REBOOT, 5, 0, is_link_up, fanout, fanout_port),
+                             "Interface {} on {} is still down after {}sec".format(fanout_port, fanout.hostname, MAX_TIME_TO_REBOOT))
+            else:
+                # Check every interfaces are down on this host every 5 sec until device boots up
+                pytest_assert(wait_until(MAX_TIME_TO_REBOOT, 5, 0, is_link_down, fanout, fanout_port),
+                             "Interface {} on {} is still up after {}sec".format(fanout_port, fanout.hostname, MAX_TIME_TO_REBOOT))
+                             
+        logger.info("All interfaces on {} are {}".format(fanout.hostname, "up" if up else "down"))
+    return True
+
+def link_status_on_all_LC(duthosts, localhost, fanouts_and_ports, up = True):
+    """
+    Return:
+        True: all links on all LCs are down
+    """
+    for LC in duthosts.frontend_nodes:
+        link_status_on_host(LC, localhost, fanouts_and_ports, up)
+    logger.info("All interfaces on all linecards are down!")
+    return True    
+
+def check_interfaces_and_services_all_LCs(duthosts, conn_graph_facts, xcvr_skip_list):
+    for LC in duthosts.frontend_nodes:
+        check_interfaces_and_services(LC, conn_graph_facts["device_conn"][LC.hostname], xcvr_skip_list)
+
+
+def test_link_down_on_sup_reboot(duthosts, localhost, enum_supervisor_dut_hostname, 
+                                conn_graph_facts, duts_running_config_facts, 
+                                fanouthosts, tbinfo, xcvr_skip_list):
+    if len(duthosts.nodes) == 1:
+        pytest.skip("Skip single-host dut for this test")
+
+    if 't2' not in tbinfo['topo']['name']:
```

Comment:
no need for this, as the test marker says for T2

Reply:
@arlakshm I'm thinking of keeping this because the test can logically work for all platforms, and maybe in the future testmarker will be changed?